### PR TITLE
[Lock] Add return types on functions in Lock Component

### PR DESCRIPTION
**Pull request by GaneshChandrasekaran-zz**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes  
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Adding return types to the classes within the Lock Component for consistency.

### CODE REVIEW

**Comment by GaneshChandrasekaran** [comment]
Ping @jderusse for thoughts/review.

File: src/Symfony/Component/Lock/Tests/LockTest.php [hunk 1]
```diff
@@ -173,7 +198,7 @@ public function testNoAutoReleaseWhenNotConfigured()
 
         $store
             ->method('exists')
-            ->willReturnOnConsecutiveCalls(array(true, false))
+            ->will($this->onConsecutiveCalls(true, false))
```
Comment (stof) [inline src/Symfony/Component/Lock/Tests/LockTest.php]: Why changing this ?

File: src/Symfony/Component/Lock/StoreInterface.php [hunk 2]
```diff
@@ -39,7 +39,7 @@ public function save(Key $key);
      * @throws LockConflictedException
      * @throws NotSupportedException
      */
-    public function waitAndSave(Key $key);
+    public function waitAndSave(Key $key): void;
```
Comment (stof) [inline src/Symfony/Component/Lock/StoreInterface.php]: Adding return types in an interface is a BC break, and so cannot be done until 5.0

Comment (GaneshChandrasekaran) [reply src/Symfony/Component/Lock/StoreInterface.php]: @stof if I didn't make changes to interfaces but only on concrete implementations, will that help get these changes out faster? And I can put up a separate PR for `5.0` version.

**Comment by nicolas-grekas** [comment]
Closing as this is a BC break for any classes extending the updated files.
See https://symfony.com/bc for more background on this.
That's maybe something we'll be able to do on Symfony 5, but that's too early to consider.
Thanks for proposing.

**Pull request closed without merging**